packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -55,10 +60,410 @@ enum _ScaffoldSlot {
   statusBar,
 }
 
-/// Geometry information for [Scaffold] components.
+/// An object that defines a position for the [FloatingActionButton]
+/// based on the [Scaffold]'s [ScaffoldPrelayoutGeometry].
+/// 
+/// Flutter provides [FloatingActionButtonPositioner]s for the common
+/// [FloatingActionButton] placements in Material Design applications. These
+/// positioners are available as static members of this class.
+/// 
+/// See also:
+/// 
+///  * [FloatingActionButton], which is a circular button typically shown in the
+///    bottom right corner of the app.
+///  * [FloatingActionButtonAnimator], which is used to animate the
+///    [Scaffold.floatingActionButton] from one [FloatingActionButtonPositioner] to 
+///    another.
+///  * [ScaffoldPrelayoutGeometry], the geometry that 
+///    [FloatingActionButtonPositioner]s use to position the [FloatingActionButton].
+abstract class FloatingActionButtonPositioner {
```

Comment:
scaffold.dart is pretty big, could this class (et al) live in fab_animation.dart?